azure-pipelines.yml

```diff
@@ -349,9 +349,34 @@ jobs:
       pr.num: $[ variables['System.PullRequest.PullRequestNumber'] ]
       pr.branch: $[ variables['System.PullRequest.SourceBranch'] ]
     steps:
-      - checkout: self
+      # some change in Azure configuration makes this fail recently (2020-01).
+      # Azure runs PR builds not on the PR commit, but on the GitHub-provided
+      # commit that would be the result of merging the PR. Recently, it looks
+      # like when it reaches the point of running this job (which has to run
+      # after the macOS one, which sometimes takes up to an hour), if master
+      # has changed in the meantime, Azure cannot find the commit it wants to
+      # build anymore. Therefore, we tell it not to checkout anything, and
+      # manually checkout the PR commit.
+      - checkout: none
       - bash: |
           set -euo pipefail
+          # Note: this is going to get the current master commit, not the
+          # result of the merge (i.e. this is not using the same commit as the
+          # other jobs in this build). This seems good enough here as all we
+          # really want is the dev-env, and conflicts should be pretty rare
```

Comment:
If this keeps being a problem, maybe we can remove `jq` and `gcloud` which don’t seem to be necessary to get the rerun button to work in case of cancellations and I don’t think recording build data is crucial.